tensorflow/python/training/learning_rate_decay_test.py

```diff
@@ -215,5 +217,83 @@ def testBeyondEndWithCycle(self):
       self.assertAllClose(decayed_lr.eval(), expected, 1e-6)
 
 
+class ExponentialDecayTest(test_util.TensorFlowTestCase):
+
+  def testDecay(self):
+    initial_lr = 0.1
+    k = 10
+    decay_rate = 0.96
+    step = state_ops.variable_op([], dtypes.int32)
+    assign = state_ops.assign(step, 0)
+    add = state_ops.assign_add(step, 1)
+    with self.test_session():
+      assign.op.run()
+      for i in range(k+1):
+        decayed_lr = learning_rate_decay.natural_exp_decay(initial_lr, step,
+                                                           k, decay_rate)
+        self.assertAllClose(initial_lr * math.exp(-i / k * decay_rate),
+                            decayed_lr.eval())
+        add.op.run()
+
+  def testStaircase(self):
+    initial_lr = 0.1
+    k = 10
+    decay_rate = 0.96
+    step = state_ops.variable_op([], dtypes.int32)
+    assign_step = state_ops.assign(step, 0)
+    increment_step = state_ops.assign_add(step, 1)
+    decayed_lr = learning_rate_decay.natural_exp_decay(initial_lr,
+                                                       step,
+                                                       k,
+                                                       decay_rate,
+                                                       staircase=True)
+    with self.test_session():
```

Comment:
@vrv apolgies for the confusion.

On Sat, Jul 9, 2016, 4:36 PM Alan Wu <alan.c.wu@gmail.com> wrote:

> @vrv let me check the code again..   I think i didn't change it for all of
> the tests, but only for test_staircase..  I'll get back to you when i am
> back on my laptop.
>
> On Sat, Jul 9, 2016, 4:27 PM Alan Wu <alan.c.wu@gmail.com> wrote:
>
>> @vrv oops, i meant as a force push...
>>
>> Is this what you mean by moving the decay rate outside?
>>
>> Thanks!
>>
>> On Sat, Jul 9, 2016, 4:17 PM Vijay Vasudevan <notifications@github.com>
>> wrote:
>>
>>> In tensorflow/python/training/learning_rate_decay_test.py
>>> <https://github.com/tensorflow/tensorflow/pull/2468#discussion_r70173417>
>>> :
>>>
>>> > +                            decayed_lr.eval())
>>> > +        add.op.run()
>>> > +
>>> > +  def testStaircase(self):
>>> > +    initial_lr = 0.1
>>> > +    k = 10
>>> > +    decay_rate = 0.96
>>> > +    step = state_ops.variable_op([], dtypes.int32)
>>> > +    assign_step = state_ops.assign(step, 0)
>>> > +    increment_step = state_ops.assign_add(step, 1)
>>> > +    decayed_lr = learning_rate_decay.natural_exp_decay(initial_lr,
>>> > +                                                       step,
>>> > +                                                       k,
>>> > +                                                       decay_rate,
>>> > +                                                       staircase=True)
>>> > +    with self.test_session():
>>>
>>> Hm, I don't see a new pull request! :(
>>>
>>> —
>>> You are receiving this because you authored the thread.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/tensorflow/tensorflow/pull/2468/files/8de656676ff49aebd3fa9e592a839127b33da93c#r70173417>,
>>> or mute the thread
>>> <https://github.com/notifications/unsubscribe/ADzDDPoLNnwggieV1kg_Cu4iRJlFWvsVks5qUCv9gaJpZM4IkGgi>
>>> .
>>>
>>